### PR TITLE
[Merged by Bors] - feat: `move-decls` prints more informative message for nameless instances

### PR DESCRIPTION
**Pull request by adomani**
Fixes a suboptimal reporting of nameless instances in `move-decls`: nameless declarations get the whole line where they are defined printed, excluding `where`/`:=`.

Reported in #12862.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — APPROVED** [review]
maintainer merge

File: scripts/no_lost_declarations.sh [hunk 1]
```diff
@@ -58,18 +58,29 @@ then
...
   echo "${full_output}" |
-  awk '{ acc[$7]=acc[$7] $4 } END{
+  awk '{
+    if(($6 == "`instance") && !($7 ~ /^[a-zA-Z]/)) {
+      rest="instance"
+      for(i=7; i<=NF-1; i++){rest=rest" "$i}
+      gsub(/`/, "", rest)
+      gsub(/ where.*/, "", rest)
+      gsub(/ :=.*/, "", rest)
+      acc[rest]=acc[rest] $4
```
Comment (jcommelin) [inline scripts/no_lost_declarations.sh:68]: Can you please add some comments what this is doing?

File: scripts/no_lost_declarations.sh [hunk 2]
```diff
@@ -58,18 +58,29 @@ then
...
+    } else { acc[$7]=acc[$7] $4 } } END{
     for(d in acc) {
       if ((acc[d] == "`+``-`") || (acc[d] == "`-``+`")) {
         printf "" } else { printf("%s %s\n", acc[d], d)
       }
     }
   }' |
   sort | uniq -c | grep -v "^ *2 " |
-  grep '\(`+`\|`-`\)' | sed 's=^ *1 =* =; s=``==g'
+  grep '\(`+`\|`-`\)' | sed 's=^ *1 ==; s=``==g; s=` = =; s=$=`='
```
Comment (jcommelin) [inline scripts/no_lost_declarations.sh:77]: This grep|sed is also becoming hard to parse quickly. Can you please add a comment?

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
bors d+
